### ISSUE TITLE
Be careful with text from the `Text` interface.

### ISSUE THREAD

**Issue by TTWNO** [odilia-app/odilia#161]
Text returned from the `org.a11y.atspi.Text` interface is _not just a String as per Rust rules_.

It contains additional contracts like that of how the [object replacement character](https://www.fileformat.info/info/unicode/char/fffc/index.htm) should be interpreted and how to interpret it.

These contracts should be reflected in the API, even if that means using a newtype. Technically, most of the usefulness of this is for screen readers (AT-SPI consumers), and is not directly related to any work done for providers, but it will change the public API for using the interfaces from the `atspi-proxies` crate (both sending and receiving).

**Comment by luukvanderduim** [comment]
I understand it a bit better now. Thanks!
Also I am happy to see this issue moved from atspi to Odilia because I suspect it is better solved in either the screen-reader or the TTS layer.

@TTWNO 
> Text returned from the `org.a11y.atspi.Text` interface is _not just a String as per Rust rules_.

The object replacement character is valid UTF-8, so it is a valid Rust string. Otherwise it could not have been constructed.

> 
> It contains additional contracts like that of how the [object replacement character](https://www.fileformat.info/info/unicode/char/fffc/index.htm) should be interpreted and how to interpret it.
> 

Is the 'object replacement character' (ORC) the only character we know of that imposes a 'contract'?
It would be nice to exclude more surprises. Ask the firefox devs if there are more practices we should know of?

So if Odilia encounters a Text object, this may be a structured Text, like a tree.
What would be the rules? 
Is the object the ORC replaces known to be the next (Text-) sibling to the current object?
You would need to keep track of lists of indexes that correspond to current objects. (I thought of ranges first, but sibling can be anything they want, not necessarily text objects)

First find out the exact rules, then find a structure to fit the rule? The above assumptions are worth nothing if wrong.
Apart from browsers, where else is this seen?

**Comment by TTWNO** [comment]
> I believe that adding such a type with the logic I mentioned above directly to atspi is beneficial because other assistive technologies may want to treat embedded links and such the way we do

It can be its own crate once we have a nice little module of code, but I don't even know which current atspi module would event apply.

> Is the 'object replacement character' (ORC) the only character we know of that imposes a 'contract'?

Yes. Confirmed by asking Matthias Clasen. Although it should probably be spec'ed somewhere.

> The above assumptions are worth nothing if wrong.

True!

> Apart from browsers, where else is this seen?

Not sure if its relevant. The point is it exists; it shouldn't matter where it is.
